service/telemetry/telemetry.go

```diff
@@ -78,16 +77,23 @@ func newLogger(cfg LogsConfig, options []zap.Option) (*zap.Logger, error) {
 	if err != nil {
 		return nil, err
 	}
+	if cfg.Sampling != nil && cfg.Sampling.Enabled {
+		logger = newSampledLogger(logger, cfg.Sampling)
+	}
 
 	return logger, nil
 }
 
-func toSamplingConfig(sc *LogsSamplingConfig) *zap.SamplingConfig {
-	if sc == nil {
-		return nil
-	}
-	return &zap.SamplingConfig{
-		Initial:    sc.Initial,
-		Thereafter: sc.Thereafter,
-	}
+func newSampledLogger(logger *zap.Logger, sc *LogsSamplingConfig) *zap.Logger {
+	// Create a logger that samples all messages to sc.Tick per second initially,
+	// and sc.Initial/sc.Thereafter of messages after that.
```

Comment:
The comment is a bit confusing. Should be something like this

```suggestion
	// Create a logger that samples every Nth message after the first M messages every S seconds 
	// where N = sc.Thereafter, M = sc.Initial, S = sc.Tick.
```